Rozdział 15/cw2.html

```diff
@@ -0,0 +1,138 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Document</title>
+</head>
+<body>
+    <canvas id="canvas" height="400" width="400"></canvas>
+    <script src="https://code.jquery.com/jquery-2.1.0.js"></script>
+
+    <script>
+      var canvas = document.getElementById("canvas");
+      var ctx = canvas.getContext("2d");
+      var width = canvas.width;
+      var height = canvas.height;
+      
+
+      var circle = function (x, y, radius, fillCircle) {
+          ctx.beginPath();
+          ctx.arc(x, y, radius, 0, Math.PI * 2, false);
+          if (fillCircle) {
+              ctx.fill();
+          } else {
+              ctx.stroke();
+          }
+      };
+
+      var Ball = function () {
+          this.x = width / 2;
+          this.y = height / 2;
+          this.xSpeed = 1;
+          this.ySpeed = 0;
+          this.speed = 5;
+          this.radius = 10;
+      };
+
+      Ball.prototype.move = function () {
+          this.x += this.xSpeed * this.speed;
+          this.y += this.ySpeed * this.speed;
+      };
+
+      Ball.prototype.draw = function () {
+          circle(this.x, this.y, this.radius, true);
+      };
+
+      Ball.prototype.setDirection = function (direction) {
+        if (direction === "up") {
+            this.xSpeed = 0;
+            this.ySpeed = -1;
+        } else if (direction === "down") {
+            this.xSpeed = 0;
+            this.ySpeed = 1;
+        } else if (direction === "left") {
+            this.xSpeed = -1;
+            this.ySpeed = 0;
+        } else if (direction === "right") {
+            this.xSpeed = 1;
+            this.ySpeed = 0;
+        } else if (direction === "stop") {
+            this.xSpeed = 0;
+            this.ySpeed = 0;
+        }
+      };
+
+      Ball.prototype.checkCollision = function () {
+        if (this.x < 0 || this.x > width) {
+          this.xSpeed = -this.xSpeed;
+        }
+        if (this.y < 0 || this.y > height) {
+          this.ySpeed = -this.ySpeed;
+        }
+      }
+
+      var ball = new Ball();
+
+      var keyActions = {
+          32: "stop",
+          37: "left",
+          38: "up",
+          39: "right",
+          40: "down"
+      };
+
+      var speeds = {
+          90: "faster",
```

Comment:
w takich przypadkach jak tutaj bym dawał komentarz co to jest za klawisz, bo to ciężko wyciągnąć z kontekstu

Reply:
np musiałem wejść tutaj https://www.oreilly.com/library/view/javascript-dhtml/9780596514082/apb.html żeby zobaczyć jak obsługiwać te rzeczy